maths/extended_gcd.nim

```diff
@@ -0,0 +1,76 @@
+## Extended Euclidean algorithm
+##
+## Provides the implementation of the function extendedGCD*
+
+import std/math
+
+
+func updateCoefficients(t0, t1, q: int): (int, int) =
+  (t1, t0 - q * t1)
+
+
+func euclidIteration(inA, inB: int): (int, int, int) =
+  assert inA >= 0 and inB >= 0
```

Comment:
If they should be positive, why use `int` here?

Reply:
Please have a look at 1faa670bef060aa73fa61ef43ebd02949cf79d70. The advantage is, that the resulting `gcd` can then also be a `uint`. The disadvantage:
- all of the _ugly casts_ - not so ugly as I thought at first,
- possible overflow in the future - ok, now `euclidIteration` is just called from `extendedGCD*`.

Please let me know about your opinion here.

Reply:
Two casts at each call are better than this assert. This raises an AssertionDefect at runtime and the user doesn't know why. With types, the error is caught at compile-time.